treq/auth.py

```diff
@@ -26,6 +182,112 @@ def request(self, method, uri, headers=None, bodyProducer=None):
             method, uri, headers=headers, bodyProducer=bodyProducer)
 
 
+class _RequestDigestAuthenticationAgent(object):
+
+    digest_auth_cache = {}
+
+    def __init__(self, agent, username, password):
+        self._agent = agent
+        self.username = username
+        self.password = password
```

Comment:
These shouldn't be public, they don't need to be if you make `build_digest_authentication_header` a private method on `_RequestDigestAuthenticationAgent`.